torch/_dynamo/utils.py

```diff
@@ -397,6 +397,9 @@ def clone_tensor(x):
 
 def clone_input(x):
     """copy while preserving strides"""
+    if isinstance(x, torch._subclasses.FakeTensor):
+        # this func fails on fake tensors in __torch_dispatch__
+        return x
```

Comment:
nuh uh, this is wrong. For fake tensor it may be acceptable to just .detach(), as a data copy is not necessary; it depends on whether or not you want the storage to be distinct or not. But better to not overload `clone_input` to work on both fake and real, IMO

Reply:
Yeah this is a placeholder while I get the rest of the stuff in shape. Will ping you about it in chat.